test/util/client/azure/compute.go

```diff
@@ -0,0 +1,356 @@
+	if err != nil {
+		return nil, err
+	}
+
+	var vms []compute.VirtualMachineScaleSetVM
+	for vmPages.NotDone() {
+		vms = append(vms, vmPages.Values()...)
+		err = vmPages.Next()
+		if err != nil {
+			return nil, err
+		}
+	}
+	return vms, nil
+}
+
+// ScaleSet returns a VirtualMachineScaleSet with a given name within a resource group
+func (az *Client) ScaleSet(resourceGroup, scaleSetName string) (*compute.VirtualMachineScaleSet, error) {
+	ss, err := az.ssc.Get(az.ctx, resourceGroup, scaleSetName)
+	if err != nil {
+		return nil, err
+	}
+	return &ss, nil
+}
+
+// ScaleOutScaleSet scales out the capacity of a given scale set to count VMs
+func (az *Client) ScaleOutScaleSet(scaleSetName string, count int) error {
+	var err error
+	scaleSet, err := az.ScaleSet(az.resourceGroup, scaleSetName)
+	if err != nil {
+		az.logger.Errorf("failed to get scale set %s: %v", scaleSetName, err)
+		return err
+	}
+	currentCapacity := *scaleSet.Sku.Capacity
+	if int64(count) < currentCapacity {
+		msg := fmt.Sprintf("scale out requires a vm count higher than the current: current=%d, requested=%d", currentCapacity, count)
+		az.logger.Errorf(msg)
+		err = fmt.Errorf(msg)
+		return err
+	}
+	az.logger.Debugf("scaling out %s scale set to %d vms", scaleSetName, count)
+	future, err := az.ssc.Update(az.ctx, az.resourceGroup, scaleSetName, compute.VirtualMachineScaleSetUpdate{
+		Sku: &compute.Sku{
+			Capacity: to.Int64Ptr(int64(count)),
+		},
+	})
+	if err != nil {
+		az.logger.Errorf("failed to scale out scale set %s: %v", scaleSetName, err)
+		return err
+	}
+	if err := future.WaitForCompletionRef(az.ctx, az.ssc.Client()); err != nil {
+		return err
+	}
+	return nil
+}
+
+// ScaleInScaleSet scales in the capacity of a given scale set to count VMs
+func (az *Client) ScaleInScaleSet(scaleSetName string, count int) error {
+	var err error
+	scaleSet, err := az.ScaleSet(az.resourceGroup, scaleSetName)
+	if err != nil {
+		az.logger.Errorf("failed to get scale set %s: %v", scaleSetName, err)
+		return err
+	}
+	currentCapacity := *scaleSet.Sku.Capacity
+	if int64(count) > currentCapacity {
+		msg := fmt.Sprintf("scale in requires a vm count lower than the current: current=%d, requested=%d", currentCapacity, count)
+		az.logger.Errorf(msg)
+		err = fmt.Errorf(msg)
+		return err
+	}
+	az.logger.Debugf("scaling in %s scale set to %d vms", scaleSetName, count)
+	future, err := az.ssc.Update(az.ctx, az.resourceGroup, scaleSetName, compute.VirtualMachineScaleSetUpdate{
+		Sku: &compute.Sku{
+			Capacity: to.Int64Ptr(int64(count)),
+		},
+	})
+	if err != nil {
+		az.logger.Errorf("failed to scale in scale set %s: %v", scaleSetName, err)
+		return err
+	}
+	if err := future.WaitForCompletionRef(az.ctx, az.ssc.Client()); err != nil {
+		return err
+	}
+	return nil
+}
+
+// UpdateScaleSetsCapacity returns a slice of the errors it encounters as it attempts to increment the capacity
+// (by one) for all scale sets within a given resource group
```

Comment:
Not sure what's the utility of this function. If you are writting the scale up/down tests you should just be scaling up/down compute.

Reply:
Nevermind, this is part of the permissions tests.